### PR TITLE
update to use ipns validator and new libp2p code

### PR DESCRIPTION
**Pull request by whyrusleeping**
still some TODOs here:
- build instructions now that it depends on go-ipfs (for the ipns validators)
- relay stuff needs advice
- fix or remove event log aggregator

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -106,17 +80,31 @@ func boostrapper() pstore.PeerInfo {
...
-	h, err := makeBasicHost(addr)
+func makeAndStartNode(ds ds.Batching, addr string, relay bool) (host.Host, *dht.IpfsDHT, error) {
+	h, err := libp2p.New(context.Background(), libp2p.ListenAddrStrings(addr))
 	if err != nil {
 		panic(err)
 	}
 
+	if relay {
+		/* TODO: fix this, need @stebalien advice
+		_, err := circuit.NewRelay(context.Background(), h, circuit.OptHop)
```
Comment (magik6k) [inline main.go]: We could also add an option to enable active relay here (so we do dialing for other peers)

File: main.go [hunk 2]
```diff
@@ -106,17 +80,31 @@ func boostrapper() pstore.PeerInfo {
...
-func makeAndStartDHT(ds ds.Batching, addr string) (host.Host, *dht.IpfsDHT, error) {
-	h, err := makeBasicHost(addr)
+func makeAndStartNode(ds ds.Batching, addr string, relay bool) (host.Host, *dht.IpfsDHT, error) {
+	h, err := libp2p.New(context.Background(), libp2p.ListenAddrStrings(addr))
 	if err != nil {
 		panic(err)
 	}
 
+	if relay {
+		/* TODO: fix this, need @stebalien advice
```
Comment (Stebalien) [inline main.go]: Pass `libp2p.EnableRelay(circuit.OptHop)` to the libp2p constructor.

**Comment by Stebalien** [comment]
We should probably extract the IPNS validator out of go-ipfs into a separate package. It's fairly self-contained.

**Comment by Stebalien** [comment]
While we're at it, I'd extract the record creation logic as well.

**Comment by whyrusleeping** [comment]
> We should probably extract the IPNS validator out of go-ipfs into a separate package. It's fairly self-contained.

Yes please!

File: main.go (hunk 2 above)
Comment (whyrusleeping) [reply main.go]: ah, super great

**Comment by Stebalien** [comment]
https://github.com/ipfs/go-ipns

**Comment by whyrusleeping** [comment]
Alright, using the new extracted ipns code.

I'm going to turn the other TODOs into help wanted issues and merge this for now. Want to get a few of them up and running :)

**Pull request merged**